src/layers/VivViewerLayer/VivViewerLayer.js

```diff
@@ -1,100 +1,38 @@
 import { CompositeLayer } from '@deck.gl/core';
 // eslint-disable-next-line import/extensions
 import VivViewerLayerBase from './VivViewerLayerBase';
-import {
-  padWithDefault,
-  isInTileBounds,
-  DEFAULT_COLOR_OFF,
-  MAX_COLOR_INTENSITY
-} from './utils';
-
-const MAX_SLIDERS_AND_CHANNELS = 6;
+import { isInTileBounds } from './utils';
+import { padColorsAndSliders } from '../utils';
 
 export default class VivViewerLayer extends CompositeLayer {
-  initializeState() {
-    this.state = {
-      imageWidth: 0,
-      imageHeight: 0,
-      tileSize: 0,
-      minZoom: 0
-    };
-  }
-
   // see https://github.com/uber/deck.gl/blob/master/docs/api-reference/layer.md#shouldupdatestate
   // eslint-disable-next-line class-methods-use-this
   shouldUpdateState({ changeFlags }) {
     return changeFlags.somethingChanged;
   }
 
-  updateState() {
```

Comment:
Don't think this stuff needs to live in state...

Reply:
I'm all for removing needed logic!